source/tools/MatchBin.h

```diff
@@ -0,0 +1,241 @@
+/**
+ *  @note This file is part of Empirical, https://github.com/devosoft/Empirical
+ *  @copyright Copyright (C) Michigan State University, MIT Software license; see doc/LICENSE.md
+ *  @date 2019
+ *
+ *  @file MatchBin.h
+ *  @brief A container that supports flexible tag-based lookup. .
+ *
+ */
+
+
+#ifndef EMP_MATCH_BIN_H
+#define EMP_MATCH_BIN_H
+
+#include <iostream>
+#include <unordered_map>
+#include <unordered_set>
+#include <functional>
+#include <algorithm>
+#include <stdexcept>
+
+#include "../base/assert.h"
+#include "../base/vector.h"
+#include "../tools/IndexMap.h"
+
+namespace emp {
+
+  template<size_t width>
+  struct HammingDistance{
+    static double Metric(BitSet<width> a, BitSet<width> b){return (a^b).CountOnes();}
+  };
+
+  struct Difference{
+    static double Metric(int a, int b){ return (double)abs(a-b);}
+  };
+
+  template<int numerator, int denominator>
```

Comment:
It should be functionally equivalent to what we're doing here, but I think using [`std::ratio`](http://www.cplusplus.com/reference/ratio/ratio/) would better express our intent/be more standard.